pilot/pkg/networking/core/v1alpha3/listener_test.go

```diff
@@ -564,6 +572,71 @@ func TestOutboundListenerForHeadlessServices(t *testing.T) {
 	}
 }
 
+func TestOutboundListenerForExternalServices(t *testing.T) {
+	svc := buildServiceWithPort("test.com", 9999, protocol.TCP, tnow)
+	svc.Attributes.ServiceRegistry = provider.Kubernetes
+
+	autoSvc := buildServiceWithPort("test.com", 9999, protocol.Unsupported, tnow)
+	autoSvc.Attributes.ServiceRegistry = provider.External
+
+	extSvc := buildServiceWithPort("example1.com", 9999, protocol.TCP, tnow)
+	extSvc.Attributes.ServiceRegistry = provider.External
+
+	tests := []struct {
+		name        string
+		instances   []*model.ServiceInstance
+		services    []*model.Service
+		listenersOn string
+	}{
+		{
+			name: "internal k8s service with ipv4 & ipv6 endpoint for Kubernetes TCP protocol",
+			instances: []*model.ServiceInstance{
+				buildServiceInstance(svc, "10.10.10.10"),
+				buildServiceInstance(svc, "fd00:10:244:1::11"),
+			},
+			services:    []*model.Service{svc},
+			listenersOn: "0.0.0.0_9999",
+		},
+		{
+			name: "external service with ipv4 & ipv6 endpoints for Kubernetes auto protocol",
+			instances: []*model.ServiceInstance{
+				buildServiceInstance(autoSvc, "10.10.10.10"),
+				buildServiceInstance(autoSvc, "fd00:10:244:1::11"),
+			},
+			services:    []*model.Service{autoSvc},
+			listenersOn: "::_9999",
+		},
+		{
+			name: "external service with ipv4 & ipv6 endpoints for Kubernetes TCP protocol",
+			instances: []*model.ServiceInstance{
+				buildServiceInstance(extSvc, "10.10.10.10"),
+				buildServiceInstance(extSvc, "fd00:10:244:1::11"),
+			},
+			services:    []*model.Service{extSvc},
+			listenersOn: "::_9999",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cg := NewConfigGenTest(t, TestOptions{
+				Services:  tt.services,
+				Instances: tt.instances,
+			})
+
+			proxy := cg.SetupProxy(getIPv6Proxy())
+
+			listeners := NewListenerBuilder(proxy, cg.env.PushContext).buildSidecarOutboundListeners(proxy, cg.env.PushContext)
+			for _, l := range listeners {
+				if l.Address.GetSocketAddress().GetPortValue() == 9999 {
```

Comment:
Is there a way to also check `GetSocketValue` or similar?

Reply:
Okay, will check it